app/assets/javascripts/directives/views/historyMenu.ts

```diff
@@ -0,0 +1,184 @@
+import { WebDirective } from '../../types';
+import { WebApplication } from '@/ui_models/application';
+import template from '%/directives/history-menu.pug';
+import { SNItem, ItemHistoryEntry } from '@node_modules/snjs/dist/@types';
+import { PureViewCtrl } from '@/views';
+import { ItemSessionHistory } from 'snjs/dist/@types/services/history/session/item_session_history';
+import { RemoteHistoryList, RemoteHistoryListEntry } from 'snjs/dist/@types/services/history/history_manager';
+import { confirmDialog } from '@/services/alertService';
+
+interface HistoryScope {
+  application: WebApplication
+  item: SNItem
+}
+
+class HistoryMenuCtrl extends PureViewCtrl implements HistoryScope {
```

Comment:
PureViewCtrl allows you to pass a type for strict checking of state: https://github.com/standardnotes/web/blob/004/app/assets/javascripts/views/abstract/pure_view_ctrl.ts#L7

Example here: https://github.com/standardnotes/web/blob/004/app/assets/javascripts/views/editor/editor_view.ts#L99

This will make it so that if you try to setState of a property not explicitly pre-defined, TS will throw an error.